etc/spack/defaults/config.yaml

```
@@ -17,7 +17,7 @@ config:
   # This is the path to the root of the Spack install tree.
   # You can use $spack here to refer to the root of the spack instance.
   install_tree:
-    root: $spack/opt/spack
+    root: $default_install_root
```

All of the config defaults in this file that use "complex" values should have comments explaining to the user what these magic variables are likely to mean for them

---

I took a crack at it for this one - is that what you're looking for?

---

I think one of the important things here is that someone reading the config can use it to figure out where the files were written. Maybe something like:
```suggestion
    # If set to a non-default value, that setting has precedence.
    # This default value tells spack to check (in order of precedence):
    # $SPACK_DATA_HOME/spack
    # $XDG_DATA_HOME/spack
    # $spack/opt/spack if previously used by an older version of spack
    # $spack/opt/data
    root: $default_install_root
```

---

I think this is addressed now